test/bestrefs/test_bestrefs.py

```diff
@@ -0,0 +1,556 @@
+import pytest
+import os, os.path
+import sys
+import re
+import datetime
+import crds
+import logging
+from crds.hst import locate
+from crds.core import log, config, utils, timestamp, cmdline, heavy_client
+from crds.hst import TYPES, INSTRUMENTS, FILEKINDS, EXTENSIONS, INSTRUMENT_FIXERS, TYPE_FIXERS
+from crds.client import api
+from crds.bestrefs import bestrefs as br
+from crds.core.timestamp import format_date, parse_date
+from astropy.time import Time
+
+
+@pytest.mark.parametrize('line, expected',
+                         [
+                             (None, None),
+                             ('auto', 'auto'),
+                             ('Mar 21 2001 12:00:00 am', '2001-03-21 00:00:00'),
+                             ('Dec 01 1993 00:00:00 UT', '1993-12-01 00:00:00'),
+                             ('Feb 08 2006 01:02AM', '2006-02-08 01:02:00'),
+                             ('12/21/1999 05:42:35', '1999-12-21 05:42:35'),
+                             ('1999-12-21T05:42:35', '1999-12-21 05:42:35'),
+                             ('12-21-1999 05:42', '1999-12-21 05:42:00'),
+                             ('19970114:053714', '1997-01-14 05:37:14'),
+                         ])
+def test_reformat_date_or_auto(line, expected):
+    """Test should return date if argument is none, 'auto' if arg is a variation of 'auto',
+    or a reformatted date otherwise."""
+    assert br.reformat_date_or_auto(line) == expected
+    log.reset()
```

Comment:
Does every test ends with `log.reset()`?

This might be a good opportunity for an [auto-used fixture](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request)

Adding something like the following should make every pytest function in this file automatically reset the log after running.
```python
@pytest.fixture(autouse=True)
def reset_log():
    yield
    log.reset()
```

If this doesn't apply to all functions, removing `autouse=True` and adding it as a function argument should work.

Reply:
Thanks, for the most part it would apply to all tests, and occasionally one or more times mid test. I'll try it out.